### PR TITLE
[PLAT-687] Propagate region from archiver source node to archiver target node

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

To confirm that nodes keep their storage region when archived

## Changes

- add single test with docs.

## QA Notes

No QA, for peer review run test at `api_tests/regions/test_models.py`

## Documentation

No docs, just code comments.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/projects/PLAT/issues/PLAT-687

### CODE REVIEW

File: api_tests/regions/test_models.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        :return:
+        """
+        schema = MetaSchema.objects.first()
+        project_with_different_regions.register_node(schema, Auth(user=user), '41-33')
+
+        regs = Registration.objects.all()
+
+        # All registrations should have the same region as the node they are registered from.
+        assert all(reg.registered_from.get_addon('osfstorage').region ==
+                   reg.get_addon('osfstorage').region for reg in regs)
```
Comment (pattisdr) [inline api_tests/regions/test_models.py]: Double indent?

File: api_tests/regions/test_models.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    def project_with_different_regions(self, user):
+        """
+        A complex project configuration with many regions.
+        :param user:
+        :return:
+        """
+        parent_node = root_node = ProjectFactory(creator=user)
+
+        # components nested five deep
+        for _ in range(0, 5):
```
Comment (pattisdr) [inline api_tests/regions/test_models.py]: Thorough 👍

**Comment by pattisdr** [comment]
Nice test @Johnetordoff, minor 5-sec cleanup

**Comment by caseyrollins** [comment]
@Johnetordoff, when I run this test it passes, but I'm not sure the `RegionFactory` is behaving as expected. I logged some info about the nodes created in the fixture and the registrations created in the test and this is what I saw:

```
Node eu7bd has region N. Virginia (us-east-1)
Node rmh9g has region N. Virginia (us-east-1)
Node et85n has region N. Virginia (us-east-1)
Node fv6hw has region N. Virginia (us-east-1)
Node jcn6b has region N. Virginia (us-east-1)
Node 9vm2e has region N. Virginia (us-east-1)
Node tpswf has region N. Virginia (us-east-1)
Node pbjh4 has region N. Virginia (us-east-1)

Registration ptm2e (registered from: pbjh4) has region N. Virginia (us-east-1)
Registration 9pujc (registered from: eu7bd) has region N. Virginia (us-east-1)
Registration sdqax (registered from: rmh9g) has region N. Virginia (us-east-1)
Registration 852g7 (registered from: tpswf) has region N. Virginia (us-east-1)
Registration dxzeq (registered from: 9vm2e) has region N. Virginia (us-east-1)
Registration 45689 (registered from: jcn6b) has region N. Virginia (us-east-1)
Registration wz9bc (registered from: fv6hw) has region N. Virginia (us-east-1)
Registration azmc5 (registered from: et85n) has region N. Virginia (us-east-1)
```

Can you look into the uniqueness of `RegionFactory`?


Also, I don't think this test belongs in the `api_tests` directory. I would suggest `osf_tests/test_region.py` instead.

**Comment by Johnetordoff** [comment]
@caseyrollins  I couldn't reproduce this, what exactly did you do to log this?

**Comment by caseyrollins** [comment]
@Johnetordoff, checked out this branch and just added a few print statements:
```
    def test_regions_stay_after_registration(self, user, project_with_different_regions):
        """
        Registering a project with components of different regions should keep those regions after registration.
        :param user:
        :param project_with_different_regions:
        :return:
        """
        from osf.models import Node
        for node in Node.objects.all():
            print 'Node {} has region {} ({})'.format(
                node._id,
                node.get_addon('osfstorage').region.name,
                node.get_addon('osfstorage').region._id
            )

        schema = MetaSchema.objects.first()
        project_with_different_regions.register_node(schema, Auth(user=user), '41-33')

        regs = Registration.objects.all()
        for reg in regs:
            print 'Registration {} (registered from: {}) has region {} ({})'.format(
                reg._id,
                reg.registered_from._id,
                reg.get_addon('osfstorage').region.name,
                reg.get_addon('osfstorage').region._id
            )

        # All registrations should have the same region as the node they are registered from.
        assert all(reg.registered_from.get_addon('osfstorage').region ==
            reg.get_addon('osfstorage').region for reg in regs)
```

**Comment by Johnetordoff** [comment]
@caseyrollins That was quite a catch! In actuality this never really worked because the new regions weren't getting saved as part of the addon AND on top of the that they weren't becoming part of the registration, so we got a false positive showing all the regions matched their nodes, because there were no unique nodes in the first place.

Anyway it should work now.

**Comment by caseyrollins** [comment]
🎉 
```
Node 8c9xt has region Region 6 (us_east_6)
Node 3ksxz has region Region 5 (us_east_5)
Node hxf39 has region Region 4 (us_east_4)
Node bcwzu has region Region 3 (us_east_3)
Node bq9j7 has region Region 2 (us_east_2)
Node pbyve has region Region 1 (us_east_1)
Node bz43p has region Region 0 (us_east_0)
Node 9pbgs has region Region 7 (us_east_7)

Registration mycra (registered from: 9pbgs) has region Region 7 (us_east_7)
Registration njy2p (registered from: 8c9xt) has region Region 6 (us_east_6)
Registration ykanz (registered from: 3ksxz) has region Region 5 (us_east_5)
Registration nucas (registered from: bz43p) has region Region 0 (us_east_0)
Registration c3dek (registered from: pbyve) has region Region 1 (us_east_1)
Registration e2xms (registered from: bq9j7) has region Region 2 (us_east_2)
Registration 8xjm4 (registered from: bcwzu) has region Region 3 (us_east_3)
Registration 8ewrx (registered from: hxf39) has region Region 4 (us_east_4)
```

I still think this test (`test_regions_stay_after_registration`) belongs in the `osf_tests` directory as opposed to the `api_tests` directory. I think the factory was fine in `addons/osfstorage/tests/factories`, is there a reason you moved it?

**Comment by Johnetordoff** [comment]
I just misread your comment about moving file, I do want to keep the factories together though, it's just easy to forget where it is.

**Pull request closed without merging**